### DISCUSSION TITLE
Новые пароли не работают

### DISCUSSION THREAD

**Discussion by aityomin** [totumonline/totum-ru-issues-and-discussions#582] [Как это сделать?] [answered]
День добрый! Возникла такая проблема. Создаю нового пользователя в  таблице "users", пытаюсь зайти, пароль не подходит. Пытаюсь создать пользователя скриптом. Скрипт, который раньше работал долго думает и выходит по таймауту.

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Непонятно что-тут сказать тк непонятно как установлено. Если это например старая установка на netangels (которая хостинг а не vps) то это может быть lock БД (там такое бывает). В этом случае просить их техподдержку его снять. 

Если VPS то рестартуйте БД:

```
service postgresql restart
```

Начиная с достаточно давно тотум автоматически ставит таймаут сброса транзакции для БД, что бы избежать таких случаев. Он равен времени исполнения скрипта (поэтому нельзя в Conf.php ставить большое время исполнения скрипта тк таким образом можно и БД загнать в блокировку). 

Ситуация возникает, когда запись в БД была инициирована и до ее завершения крашнулся сервер PHP (особенно актуально когда база на одном сервере а php на другом), в БД получается висящая транзакция. На новых версиях провисит не более таймаута.

**Reply by aityomin** [reply]
Установлен и работает уже год на собственном сервере. Сразу после обнаружения проблемы все рестартовали. Проблема не прошла.
PS вопросов к таймауту нету

**Reply by ttm-alexeypolunin** [reply]
наличие ремоутов, которые подключаются от api-пользователя раз в 10 секунд?

**Reply by ttm-alexeypolunin** [reply]
Вобщем чат пишет, что запись из таблицы pg_locks удаляется при завершении транзакции, если транзакция каким-то образом крашнулась, то блокировка может остаться висеть и после перезагрузки.

Посмотреть что висит в блокировках:

```
su postgres
```
```
psql
```
```
SELECT 
    pg_locks.pid,
    pg_locks.locktype,
    pg_class.relname AS table,
    pg_locks.objid AS row_id
FROM 
    pg_locks
JOIN 
    pg_class ON pg_locks.relation = pg_class.oid
WHERE 
    pg_locks.locktype = 'SELECT 
    pg_locks.pid,
    pg_locks.locktype,
    pg_class.relname AS table,
    pg_locks.objid AS row_id
FROM 
    pg_locks
JOIN 
    pg_class ON pg_locks.relation = pg_class.oid
WHERE 
    pg_locks.locktype = 'tuple';
```

`tuple` — это строковые блокировки.

Посмотреть активные транзакции:

```
SELECT 
    pid,
    usename,
    datname,
    state,
    query,
    xact_start,
    query_start,
    state_change
FROM 
    pg_stat_activity
WHERE 
    state = 'active'
    AND xact_start IS NOT NULL
ORDER BY 
    xact_start;
```

**Reply by aityomin** [reply]
Запросы выдают пустоту...

**Reply by ttm-alexeypolunin** [reply]
Значит что-то другое...

**Reply by aityomin** [reply]
Знать бы что...

**Reply by ttm-alexeypolunin** [reply]
Ну а как мы мы можем узнать — никак.
